File: 0238-product-of-array-except-self/0238-product-of-array-except-self.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        ans = [1] * len(nums)
        prefix_product = 1
        postfix_product = 1

        for i in range(len(nums)):
            ans[i] = prefix_product
            prefix_product *= nums[i]
        
        for i in range(len(nums) - 1, -1, -1):
            ans[i] *= postfix_product
            postfix_product *= nums[i]

        return ans

File: 0238-product-of-array-except-self/0238-product-of-array-except-self.go
func productExceptSelf(nums []int) []int {
    px := 1
    sx := 1
    ans := make([]int, len(nums))

    for i := range nums {
        ans[i] = px
        px *= nums[i]
    }

    for i := len(nums) - 1; i >= 0; i-- {
        ans[i] *= sx
        sx *= nums[i]
    }

    return ans
}